### PR TITLE
Implement Method to Get Test Session by Test Id

### PR DESCRIPTION
**Pull request by mahadzaryab1**
## Notion ticket link

[Get test sessions by test id](https://www.notion.so/uwblueprintexecs/Get-test-sessions-by-test-id-d4972e435063498680ab31d208a866f8)

## Implementation description
* Implemented a method to get test sessions based on the test ID
* Added unit tests for new method

## Steps to test
1.

## What should reviewers focus on?
* 

## Checklist
- [ ] My PR name is descriptive and in imperative tense
- [ ] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ ] I have run the appropriate linter(s)
- [ ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/services/interfaces/testSessionService.ts [hunk 1]
```diff
@@ -99,4 +99,8 @@ export interface ITestSessionService {
    * This method fetches all the test sessions from the database.
    */
   getAllTestSessions(): Promise<Array<TestSessionResponseDTO>>;
+
+  getTestSessionsByTestId(
```
Comment (carissa-tang) [inline backend/services/interfaces/testSessionService.ts:116]: nit: add documentation

File: backend/services/implementations/__tests__/testSessionService.test.ts [hunk 2]
```diff
@@ -42,4 +42,27 @@ describe("mongo testSessionService", (): void => {
...
+      assertResponseMatchesExpected(mockTestSessionsWithSameTestId[i], res[i]);
+      assertResultsResponseMatchesExpected(
+        mockTestSessionsWithSameTestId[i],
+        res[i],
+      );
+    }
+  });
+
+  it("getTestSessionsByTestId for non-existing id", async () => {
+    await MgTestSession.insertMany(mockTestSession);
```
Comment (carissa-tang) [inline backend/services/implementations/__tests__/testSessionService.test.ts]:
```suggestion
    await MgTestSession.create(mockTestSession);
```
should be this if i'm understanding correctly?

File: backend/testUtils/testSession.ts [hunk 3]
```diff
@@ -21,6 +21,27 @@ export const mockTestSession: TestSessionRequestDTO = {
   startTime: new Date("2021-09-01T09:00:00.000Z"),
 };
 
+export const mockTestSessionsWithSameTestId: Array<TestSessionRequestDTO> = [
```
Comment (carissa-tang) [inline backend/testUtils/testSession.ts:24]: we could merge this with `mockTestSession` and just add a second object. but two vars might also be more clear. i guess as long as we are consistent, we can go with either one

File: backend/services/implementations/__tests__/testSessionService.test.ts [hunk 4]
```diff
@@ -42,4 +42,27 @@ describe("mongo testSessionService", (): void => {
...
     assertResultsResponseMatchesExpected(mockTestSession, res[0]);
   });
+
+  it("getTestSessionsByTestId", async () => {
+    await MgTestSession.create(mockTestSessionsWithSameTestId);
+    const testId = "62c248c0f79d6c3c9ebbea95";
+
+    const res = await testSessionService.getTestSessionsByTestId(testId);
+    expect(res.length).toEqual(mockTestSessionsWithSameTestId.length);
+    for (let i = 0; i < res.length; i += 1) {
```
Comment (carissa-tang) [inline backend/services/implementations/__tests__/testSessionService.test.ts]:
```suggestion
    res.forEach((testSession: TestSessionResponseDTO, i) => {
```
nit: for consistency sake

**Review by carissa-tang — CHANGES_REQUESTED** [review]

File: backend/testUtils/testSession.ts (hunk 3 above)
Comment (mahadzaryab1) [reply backend/testUtils/testSession.ts:24]: yeah that was my reasoning - to keep the data for the test cases separate

**Review by joyce-shi — APPROVED** [review]
🛳

**Review by carissa-tang — APPROVED** [review]

**Pull request merged**